presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterSourceOperator.java

```diff
@@ -91,45 +93,69 @@ public OperatorFactory duplicate()
         }
     }
 
+    @Immutable
+    public static class DynamicFilterSummary
+            implements Mergeable<DynamicFilterSummary>, OperatorInfo
+    {
+        private final TupleDomain<Integer> tupleDomain;
+
+        @JsonCreator
+        public DynamicFilterSummary(@JsonProperty("tupleDomain") TupleDomain<Integer> tupleDomain)
+        {
+            this.tupleDomain = tupleDomain;
+        }
+
+        @JsonProperty
+        public TupleDomain<Integer> getTupleDomain()
+        {
+            return tupleDomain;
+        }
+
+        @Override
+        public DynamicFilterSummary mergeWith(DynamicFilterSummary other)
+        {
+            return new DynamicFilterSummary(columnWiseUnion(tupleDomain, other.tupleDomain));
+        }
+    }
+
     private static final int DEFAULT_POSITIONS_LIMIT = 1000;
 
     private final OperatorContext operatorContext;
     private final List<Type> types;
     private final List<Integer> filterChannels;
-    private final TupleDomainSource tupleDomainSource;
     private final int positionsLimit;
 
     private Page page;
     private boolean finishing;
     private long positionCount;
     private List<ImmutableList.Builder<Object>> valuesBuilder;
     private boolean[] nullsAllowed;
+    private DynamicFilterSummary summary;
 
     public DynamicFilterSourceOperator(
             OperatorContext operatorContext,
             List<Type> types,
-            List<Integer> filterChannels,
-            TupleDomainSource tupleDomainSource)
+            List<Integer> filterChannels)
     {
-        this(operatorContext, types, filterChannels, tupleDomainSource, DEFAULT_POSITIONS_LIMIT);
+        this(operatorContext, types, filterChannels, DEFAULT_POSITIONS_LIMIT);
     }
 
     public DynamicFilterSourceOperator(
             OperatorContext operatorContext,
             List<Type> types,
             List<Integer> filterChannels,
-            TupleDomainSource tupleDomainSource,
             int positionsLimit)
     {
         this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
         this.types = ImmutableList.copyOf(types);
         this.filterChannels = ImmutableList.copyOf(filterChannels);
         this.valuesBuilder = filterChannels.stream().map(channel -> ImmutableList.builder()).collect(toImmutableList());
         this.nullsAllowed = new boolean[filterChannels.size()];
-        this.tupleDomainSource = requireNonNull(tupleDomainSource, "tupleDomainSource is null");
 
         checkArgument(positionsLimit > 0, "positionsLimit must be greater than zero");
         this.positionsLimit = positionsLimit;
+
+        operatorContext.setInfoSupplier(() -> summary);
```

Comment:
It's safe to return `null` there.
Please have a look on the only use of `infoSupplier`, which is `OperatorContext.gerOperatorStats()`.
It reads:
```
Supplier<OperatorInfo> infoSupplier = this.infoSupplier.get();
OperatorInfo info = Optional.ofNullable(infoSupplier).map(Supplier::get).orElse(null);
```
Then, in other places where `info` is referred, it's always checked for nullness.